src/nvim/normal.c

```diff
@@ -1908,6 +1908,13 @@ bool do_mouse(oparg_T *oap, int c, int dir, long count, bool fixindent)
     StlClickDefinition *click_defs = in_status_line ? wp->w_status_click_defs
                                                     : wp->w_winbar_click_defs;
 
+    if (in_status_line && p_ls == 3) {
```

Comment:
```suggestion
    if (in_status_line && gloal_stl_height() > 0) {
```

Reply:
Not sure if `click_grid` also needs to be check here.

Reply:
Updated.

Is there any documentation about the underlying architecture of the Neovim's UI? I'm not sure what this grid system is 🤔

Reply:
Oh, `click_grid` must be `DEFAULT_GRID_HANDLE` if `on_status_line` is `true`, so there is no need to check `click_grid`.